frontend/src/pages/user/UserAnswerPage.js

```diff
@@ -39,13 +41,48 @@ export default function UserAnswerPage() {
   const lastPage =
     questions && Array.isArray(questions) ? questions.length + 1 : 0;
 
+  const totalQuestion =
+    questions && Array.isArray(questions) ? questions.length : 0;
+  const passingScore = 75;
+
+  useEffect(() => {
+    setPassed(scorePercentage >= passingScore ? true : false);
+  }, [scorePercentage]);
+
+  useEffect(() => {
+    const calculate = async () => {
+      try {
+        const result = await calculateScore({
+          score,
+          totalQuestion,
+          answers,
+          correctAnswers,
+        });
+        setScorePercentage(result.result);
+        setScore(result.score);
+        setIsCorrect(result.isCorrect);
+      } catch (error) {
+        console.log(error);
+      }
+    };
+    calculate();
+  }, [isDone]);
+
+  const handleAnswer = (answer, correctAnswer) => {
+    setCurrentPage(currentPage + 1);
+    setAnswers(answers.concat([answer]));
+    setCorrectAnswers(correctAnswers.concat([correctAnswer]));
```

Comment:
Dont build the correct answers on FE. You can just query it on BE for security purposes. You dont want users to modify the data